src/Core/Services/SyncService.cs

```diff
@@ -386,36 +386,41 @@ private async Task SyncSendsAsync(string userId, List<SendResponse> response)
 
         private async Task SyncPasswordlessLoginRequestsAsync(string userId)
         {
-            // if the user has not enabled passwordless logins ignore requests
-            if (!await _stateService.GetApprovePasswordlessLoginsAsync(userId))
+            try
             {
-                return;
-            }
+                // if the user has not enabled passwordless logins ignore requests
+                if (!await _stateService.GetApprovePasswordlessLoginsAsync(userId))
+                {
+                    return;
+                }
 
-            var loginRequests = await _apiService.GetAuthRequestAsync();
-            if (loginRequests == null || !loginRequests.Any())
-            {
-                return;
-            }
+                var loginRequests = await _apiService.GetAuthRequestAsync();
+                if (loginRequests == null || !loginRequests.Any())
+                {
+                    return;
+                }
 
-            var validLoginRequest = loginRequests.Where(l => l.RequestApproved == null
-                                                 && l.ResponseDate == null
-                                                 && l.CreationDate.ToUniversalTime().AddMinutes(Constants.PasswordlessNotificationTimeoutInMinutes) > DateTime.UtcNow)
-                                     .OrderByDescending(x => x.CreationDate)
-                                     .FirstOrDefault();
+                var validLoginRequest = loginRequests.Where(l => !l.IsAnswered && !l.IsExpired)
+                                         .OrderByDescending(x => x.CreationDate)
+                                         .FirstOrDefault();
 
-            if (validLoginRequest is null)
-            {
-                return;
-            }
+                if (validLoginRequest is null)
+                {
+                    return;
+                }
 
-            await _stateService.SetPasswordlessLoginNotificationAsync(new PasswordlessRequestNotification()
-            {
-                Id = validLoginRequest.Id,
-                UserId = userId
-            });
+                await _stateService.SetPasswordlessLoginNotificationAsync(new PasswordlessRequestNotification()
+                {
+                    Id = validLoginRequest.Id,
+                    UserId = userId
+                });
 
-            _messagingService.Send(Constants.PasswordlessLoginRequestKey);
+                _messagingService.Send(Constants.PasswordlessLoginRequestKey);
+            }
+            catch (Exception ex)
+            {
+                LoggerHelper.LogEvenIfCantBeResolved(ex);
```

Comment:
if it can resolve the `_stateService` and the `_messagingService` it can resolve the logger, so inject it through the interface and use the instance here instead of the static helper method